tests/metrics/collect_mem_consmd.sh

```diff
@@ -128,38 +140,100 @@ function start_collect()
 # It kills the collectl process.
 function stop_collect()
 {
+	echo "Stop collectl"
 	kill "$COLLECT_PID"
 }
 
-# Get the average about a "csv" file produced
+# It launches a instance of smem tool
+# in order to get metrics about proportional
+# set size.
+function start_pss_monitor()
+{
+	bash ./smem_monitor.sh $PROC &
+	SMEM_PID=$!
+}
+
+# It terms smem process
+function stop_pss_monitor()
+{
+	kill "$SMEM_PID"
+}
+
+# Get PSS memory metrics
+function get_pss_metrics()
+{
+	pss_sum=0
+	pss_count=0
+	samples=""
+
+	while read line; do
+		pss_mem=$(echo "$line" | sed 's/[A-Za-z]*//g')
+		res=$(echo "$pss_mem > 0" | bc )
+		if [ $res -eq 1 ];then
+			pss_sum=$(echo "$pss_sum + $pss_mem"| bc -l )
+			pss_count=$(( $pss_count + 1 ))
+			if [ ! -z "$samples" ];then
+				samples="$samples ,"
+			fi
+			samples="$samples $pss_mem"
+		fi
+	done < "$PSS_FILE"
+
+	stdev=$(python3.6 -c "import statistics; print (statistics.stdev([$samples]))")
+	avg=$(echo "$pss_sum / $pss_count" | bc -l)
+	pstdev=$(echo "($stdev / $avg) * 100" | bc -l)
+	echo "avg: $avg stdev: $stdev pstdev: $pstdev"
+}
+
+# Get metrics about a "csv" file produced
 # by collectl tool.
-function get_average()
+function get_collect_metrics()
 {
-	sum=0
-	count=0
+	data_sum=0
+	data_count=0
+	samples=""
 
 	while read line; do
 		cmd=$(echo "$line" | awk -F "," '{print $30}' | grep $PROC)
 		if [ ! -z "$cmd" ];then
 			binary="$(echo "$cmd" | cut -d " " -f1 | \
 				xargs basename -z | grep $F_OPTS $PROC)"
 			if [ ! -z "$binary" ];then
-				mem_data=$(echo "$line" | awk -F "," '{print $12}')
-				if (( $mem_data > 0 ));then
-					sum=$(( $sum + $mem_data ))
-					count=$(( $count + 1 ))
+				data_mem=$(echo "$line" | awk -v colum=$1 -F "," '{print $colum}')
```

Comment:
Another comment: this deeply indented code is very difficult to read (and thus more likely to contain errors). I'd try to simplify it by doing things like:

```
while read line; do
    # If the line doesn't match, ignore immediately.
    # This means you save 1 leve of indentation - easier to read ;-)
    [ -n "$cmd" ] || continue

    # Again, test the line and ignore it if you don't get the result you want.
    [ -n "$binary" ] || continue

    # The main code is now here, but doesn't need to be deeply nested.
   
done
```